### PR TITLE
Adding checkbox for data consent

### PR DESCRIPTION
**Pull request by kfless12**
an additional Checkbox field was added for the user to click and accept data consent. 
-the checkbox component was modified to include a "required" field in order to have an alert message pop up when the user tries to submit a form without checking the box.

-data from the checkbox was added to the users data when creating a user in firebase. the functions of firebase were updated to reflect this data change.

*note my linter seemed to not work properly until an additional line of code was added to the eslintconfig file.

### CODE REVIEW

**Comment by timstokman** [comment]
I'm ok with doing large scale formatting, but if you do that, could you not put everything together with actual changes in the code? Makes it rather hard to do an actual code review.

Could you tell me which bits actually changed? I can do a review for those.

**Comment by kfless12** [comment]
I apologize my linter wasn't configured correctly initially and it reformated everything in the frontend. I fixed the issue but never reset the formatting so the changes got pushed up. I can pull an older commit and reset the formatting to the initial settings so the changes are reviewable.

File: firebase/functions/.eslintrc.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ module.exports = {
   parser: "@typescript-eslint/parser",
   parserOptions: {
     project: ["tsconfig.json", "tsconfig.dev.json"],
+    tsconfigRootDir: __dirname,
```
Comment (kfless12) [inline firebase/functions/.eslintrc.js:18]: update to eslint file needed for my linter to work properly.

File: firebase/functions/src/index.ts [hunk 2]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+      functions.logger.debug("createUser parameters", data);
+      const [
+        email,
+        chainId,
+        name,
+        phoneNumber,
+        newsletter,
+        actionsNewsletter,
+        address,
+        dataConsent,
```
Comment (kfless12) [inline firebase/functions/src/index.ts:37]: additional variable added on line 37 to account for checkbox boolean

File: firebase/functions/src/index.ts [hunk 3]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+      ] = [
+        data.email,
+        data.chainId,
+        data.name,
+        data.phoneNumber,
+        data.newsletter,
+        data.actionsNewsletter,
+        data.address,
+        data.dataConsent,
```
Comment (kfless12) [inline firebase/functions/src/index.ts:46]: data value added on line 46 aswell

File: firebase/functions/src/index.ts [hunk 4]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+          html: verificationEmail,
+        },
+      });
+      functions.logger.debug("Adding user supplemental information to firebase");
+      await db.collection("users").doc(userRecord.uid).set({
+        chainId,
+        address,
+        newsletter,
+        actionsNewsletter,
+        dataConsent,
```
Comment (kfless12) [inline firebase/functions/src/index.ts:87]: dataConsent value added on line 87

File: firebase/functions/src/index.ts [hunk 5]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+    .https.onCall(async (data: any, context: functions.https.CallableContext) => {
+      functions.logger.debug("updateUser parameters", data);
+      const [
+        uid,
+        name,
+        phoneNumber,
+        newsletter,
+        actionsNewsletter,
+        address,
+        dataConsent,
```
Comment (kfless12) [inline firebase/functions/src/index.ts:194]: line 194 value added for data Consent

File: firebase/functions/src/index.ts [hunk 6]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+      ] = [
+        data.uid,
+        data.name,
+        data.phoneNumber,
+        data.newsletter,
+        data.actionsNewsletter,
+        data.address,
+        data.dataConsent,
```
Comment (kfless12) [inline firebase/functions/src/index.ts:202]: line 202 data consent added

File: firebase/functions/src/index.ts [hunk 7]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+          displayName: name,
+          disabled: false,
+        });
+        functions.logger.debug("updated user", userRecord);
+        await db.collection("users").doc(userRecord.uid).set(
+            {
+              address,
+              newsletter,
+              actionsNewsletter,
+              dataConsent,
```
Comment (kfless12) [inline firebase/functions/src/index.ts:217]: data consent added to update

File: firebase/functions/src/index.ts [hunk 8]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
+          uid: user.uid,
+          email: user.email,
+          name: user.displayName,
+          phoneNumber: user.phoneNumber,
+          emailVerified: user.emailVerified,
+          chainId: userData.get("chainId"),
+          address: userData.get("address"),
+          newsletter: userData.get("newsletter"),
+          actionsNewsletter: userData.get("actionsNewsletter"),
+          dataConsent: userData.get("dataConsent"),
```
Comment (kfless12) [inline firebase/functions/src/index.ts:255]: data consent added to collection function line 255

File: firebase/functions/src/index.ts [hunk 9]
```diff
@@ -22,258 +22,278 @@ function wrapInECMAPromise<T>(fn: () => Promise<T>): Promise<T> {
...
```
Comment (kfless12) [inline firebase/functions/src/index.ts:290]: line 290 data consent added

**Comment by timstokman** [comment]
Is this just adding a checkbox for "data consent" when creating a user, and then storing this value? What for? If it's for GDPR, I think there's some misunderstanding here, because that's not really how it works.

**Comment by kfless12** [comment]
The commit was updated to only reflect what was changed on my end and the code was return to its normal established format

**Comment by kfless12** [comment]
> Is this just adding a checkbox for "data consent" when creating a user, and then storing this value? What for? If it's for GDPR, I think there's some misunderstanding here, because that's not really how it works.

@giuliamml may be able to answer that question more clearly but yes It was one of the tasks on the "issue" tab that i was told to work on.

**Comment by timstokman** [comment]
I moved the discussion to slack @kfless12 . The code looks fine, but I'm not sure this is a good way to implement GDPR related functionality.

**Pull request closed without merging**